src/mbgl/map/transform.cpp

```diff
@@ -314,6 +312,10 @@ void Transform::flyTo(const CameraOptions &camera, const AnimationOptions &anima
                 util::interpolate(startEdgeInsets.right(), padding.right(), us)
             };
         }
+        auto maxPitch = getMaxPitchForEdgeInsets(state.edgeInsets);
```

Comment:
could the above statement be calculated only on `state.edgeInsets` change?

Reply:
Use of optionals mandate this implementation: if no insets or no pitch are supplied, means we use current values and still need to set and ensure validity of what is set.